sdk/dotnet/Pulumi/X/Automation/IWorkspace.cs

```diff
@@ -0,0 +1,198 @@
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Threading.Tasks;
+
+namespace Pulumi.X.Automation 
+{
+    /// <summary>
+    /// Workspace is the execution context containing a single Pulumi project, a program, and multiple stacks.
+    /// <para/>
+    /// Workspaces are used to manage the execution environment, providing various utilities such as plugin
+    /// installation, environment configuration ($PULUMI_HOME), and creation, deletion, and listing of Stacks.
+    /// </summary>
+    public interface IWorkspace 
+    {
+        /// <summary>
+        /// The working directory to run Pulumi CLI commands.
+        /// </summary>
+        string WorkDir { get; }
+
+        /// <summary>as
+        /// The directory override for CLI metadata if set.
+        /// <para/>
+        /// This customizes the location of $PULUMI_HOME where metadata is stored and plugins are installed.
+        /// </summary>
+        string? PulumiHome { get; }
+
+        /// <summary>
+        /// The secrets provider to use for encryption and decryption of stack secrets.
+        /// <para/>
+        /// See: https://www.pulumi.com/docs/intro/concepts/config/#available-encryption-providers
+        /// </summary>
+        string? SecretsProvider { get; }
+
+        /// <summary>
+        /// The inline program <see cref="PulumiFn"/> to be used for Preview/Update operations if any.
+        /// <para/>
+        /// If none is specified, the stack will refer to <see cref="ProjectSettings"/> for this information.
+        /// </summary>
+        PulumiFn? Program { get; set; }
+
+        /// <summary>
+        /// Environment values scoped to the current workspace. These will be supplied to every Pulumi command.
+        /// </summary>
+        IDictionary<string, string>? EnvironmentVariables { get; set;}
+
+        /// <summary>
+        /// Returns project settings for the current project if any.
+        /// </summary>
+        Task<ProjectSettings?> GetProjectSettingsAsync();
```

Comment:
Should we be passing a `CancellationToken` around for the asynchronous methods on `IWorkspace`? *Are* these methods cancel-able?